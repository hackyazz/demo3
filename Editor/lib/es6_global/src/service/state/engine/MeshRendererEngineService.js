

import * as MeshRendererAPI$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/api/MeshRendererAPI.js";

function setDrawMode(value, component, state) {
  return MeshRendererAPI$Wonderjs.setMeshRendererDrawMode(component, value, state);
}

var create = MeshRendererAPI$Wonderjs.createMeshRenderer;

var getDrawMode = MeshRendererAPI$Wonderjs.getMeshRendererDrawMode;

var getMeshRendererIsRender = MeshRendererAPI$Wonderjs.getMeshRendererIsRender;

var setMeshRendererIsRender = MeshRendererAPI$Wonderjs.setMeshRendererIsRender;

export {
  create ,
  getDrawMode ,
  setDrawMode ,
  getMeshRendererIsRender ,
  setMeshRendererIsRender ,
  
}
/* MeshRendererAPI-Wonderjs Not a pure module */
