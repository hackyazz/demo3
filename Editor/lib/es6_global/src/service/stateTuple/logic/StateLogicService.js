

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as StateEditorService$WonderEditor from "../../state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../state/engine/state/StateEngineService.js";
import * as DirectorEngineService$WonderEditor from "../../state/engine/DirectorEngineService.js";

function getEngineStateToGetData(handleFunc) {
  return Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0));
}

function getAndSetEngineState(handleFunc) {
  StateEngineService$WonderEditor.setState(Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function refreshEngineState(engineState) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, engineState));
  return /* () */0;
}

function refreshEngineStateAndReturnEngineState(engineState) {
  return DirectorEngineService$WonderEditor.loopBody(0, engineState);
}

function getAndRefreshEngineState(param) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function getAndRefreshEngineStateWithFunc(handleFunc) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0))));
  return /* () */0;
}

function loopBodyWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getAndRefreshEngineStateWhenStop(param) {
  StateEngineService$WonderEditor.setState(loopBodyWhenStop(StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function renderWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getEditorState(handleFunc) {
  return Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0));
}

function getAndSetEditorState(handleFunc) {
  StateEditorService$WonderEditor.setState(Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0)));
  return /* () */0;
}

function getStateToGetData(handleFunc) {
  return Curry._1(handleFunc, /* tuple */[
              StateEditorService$WonderEditor.getState(/* () */0),
              StateEngineService$WonderEditor.unsafeGetState(/* () */0)
            ]);
}

function getAndSetState(handleFunc) {
  var match = Curry._1(handleFunc, /* tuple */[
        StateEditorService$WonderEditor.getState(/* () */0),
        StateEngineService$WonderEditor.unsafeGetState(/* () */0)
      ]);
  StateEditorService$WonderEditor.setState(match[0]);
  StateEngineService$WonderEditor.setState(match[1]);
  return /* () */0;
}

function setState(param) {
  StateEditorService$WonderEditor.setState(param[0]);
  StateEngineService$WonderEditor.setState(param[1]);
  return /* () */0;
}

export {
  getEngineStateToGetData ,
  getAndSetEngineState ,
  refreshEngineState ,
  refreshEngineStateAndReturnEngineState ,
  getAndRefreshEngineState ,
  getAndRefreshEngineStateWithFunc ,
  loopBodyWhenStop ,
  getAndRefreshEngineStateWhenStop ,
  renderWhenStop ,
  getEditorState ,
  getAndSetEditorState ,
  getStateToGetData ,
  getAndSetState ,
  setState ,
  
}
/* StateEditorService-WonderEditor Not a pure module */
