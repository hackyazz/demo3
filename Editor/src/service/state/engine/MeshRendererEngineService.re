open Wonderjs;

let create = MeshRendererAPI.createMeshRenderer;

let getDrawMode = MeshRendererAPI.getMeshRendererDrawMode;

let setDrawMode = (value, component, state) =>
  MeshRendererAPI.setMeshRendererDrawMode(component, value, state);

let getMeshRendererIsRender = MeshRendererAPI.getMeshRendererIsRender;

let setMeshRendererIsRender = MeshRendererAPI.setMeshRendererIsRender;

