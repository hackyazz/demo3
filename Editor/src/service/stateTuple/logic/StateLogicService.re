let getEngineStateToGetData = handleFunc =>
  StateEngineService.unsafeGetState() |> handleFunc;

let getAndSetEngineState = handleFunc =>
  StateEngineService.unsafeGetState()
  |> handleFunc
  |> StateEngineService.setState
  |> ignore;

let refreshEngineState = engineState =>
  engineState
  |> DirectorEngineService.loopBody(0.)
  |> StateEngineService.setState
  |> ignore;

let refreshEngineStateAndReturnEngineState = engineState =>
  engineState |> DirectorEngineService.loopBody(0.);

let getAndRefreshEngineState = () =>
  StateEngineService.unsafeGetState()
  |> DirectorEngineService.loopBody(0.)
  |> StateEngineService.setState
  |> ignore;

let getAndRefreshEngineStateWithFunc = handleFunc =>
  StateEngineService.unsafeGetState()
  |> handleFunc
  |> DirectorEngineService.loopBody(0.)
  |> StateEngineService.setState
  |> ignore;

let loopBodyWhenStop = engineState =>
  StateEditorService.getIsRun() ?
    engineState : engineState |> DirectorEngineService.loopBody(0.);

let getAndRefreshEngineStateWhenStop = () =>
  StateEngineService.unsafeGetState()
  |> loopBodyWhenStop
  |> StateEngineService.setState
  |> ignore;

let renderWhenStop = engineState =>
  StateEditorService.getIsRun() ?
    engineState : engineState |> DirectorEngineService.loopBody(0.);

let getEditorState = handleFunc => StateEditorService.getState() |> handleFunc;

let getAndSetEditorState = handleFunc =>
  StateEditorService.getState()
  |> handleFunc
  |> StateEditorService.setState
  |> ignore;

let getStateToGetData = handleFunc =>
  (StateEditorService.getState(), StateEngineService.unsafeGetState())
  |> handleFunc;

let getAndSetState = handleFunc => {
  let (editorState, engineState) =
    (StateEditorService.getState(), StateEngineService.unsafeGetState())
    |> handleFunc;

  editorState |> StateEditorService.setState |> ignore;
  engineState |> StateEngineService.setState |> ignore;

  ();
};

let setState = ((editorState, engineState)) => {
  editorState |> StateEditorService.setState |> ignore;
  engineState |> StateEngineService.setState |> ignore;

  ();
};