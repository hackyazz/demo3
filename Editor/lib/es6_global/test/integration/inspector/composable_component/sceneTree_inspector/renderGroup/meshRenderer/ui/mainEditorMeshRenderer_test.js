

import * as Curry from "../../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as TestTool$WonderEditor from "../../../../../../../tool/TestTool.js";
import * as ReactTestTool$WonderEditor from "../../../../../../../tool/ReactTestTool.js";
import * as GameObjectTool$WonderEditor from "../../../../../../../tool/GameObjectTool.js";
import * as StateLogicService$WonderEditor from "../../../../../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as BuildComponentTool$WonderEditor from "../../../../../../../tool/BuildComponentTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../../../tool/MainEditorSceneTool.js";
import * as MeshRendererEngineService$WonderEditor from "../../../../../../../../src/service/state/engine/MeshRendererEngineService.js";
import * as MainEditorMeshRendererTool$WonderEditor from "../tool/MainEditorMeshRendererTool.js";

describe("mainEditor meshRenderer", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                return MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("test meshRenderer change drawMode", (function () {
                beforeEach((function () {
                        return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
                      }));
                describe("test set drawMode to Lines", (function () {
                        Wonder_jest.test("test snapshot", (function () {
                                var lineType = MainEditorMeshRendererTool$WonderEditor.getDrawModeLineType(/* () */0);
                                MainEditorMeshRendererTool$WonderEditor.changeMode(lineType, undefined, undefined, undefined, /* () */0);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMeshRenderer(TestTool$WonderEditor.buildEmptyAppState(/* () */0)));
                              }));
                        return Wonder_jest.test("test logic", (function () {
                                      var lineType = MainEditorMeshRendererTool$WonderEditor.getDrawModeLineType(/* () */0);
                                      MainEditorMeshRendererTool$WonderEditor.changeMode(lineType, undefined, undefined, undefined, /* () */0);
                                      var meshRenderer = GameObjectTool$WonderEditor.getCurrentGameObjectMeshRenderer(/* () */0);
                                      return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                            return MeshRendererEngineService$WonderEditor.getDrawMode(meshRenderer, param);
                                                          }))), lineType);
                                    }));
                      }));
                describe("test set drawMode to Points", (function () {
                        Wonder_jest.test("test snapshot", (function () {
                                var pointType = MainEditorMeshRendererTool$WonderEditor.getDrawModePointType(/* () */0);
                                MainEditorMeshRendererTool$WonderEditor.changeMode(pointType, undefined, undefined, undefined, /* () */0);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMeshRenderer(TestTool$WonderEditor.buildEmptyAppState(/* () */0)));
                              }));
                        return Wonder_jest.test("test logic", (function () {
                                      var pointType = MainEditorMeshRendererTool$WonderEditor.getDrawModePointType(/* () */0);
                                      MainEditorMeshRendererTool$WonderEditor.changeMode(pointType, undefined, undefined, undefined, /* () */0);
                                      var meshRenderer = GameObjectTool$WonderEditor.getCurrentGameObjectMeshRenderer(/* () */0);
                                      return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                            return MeshRendererEngineService$WonderEditor.getDrawMode(meshRenderer, param);
                                                          }))), pointType);
                                    }));
                      }));
                describe("test set drawMode to Triangle_fan", (function () {
                        Wonder_jest.test("test snapshot", (function () {
                                var triangleFanType = MainEditorMeshRendererTool$WonderEditor.getDrawModeTriangleFanType(/* () */0);
                                MainEditorMeshRendererTool$WonderEditor.changeMode(triangleFanType, undefined, undefined, undefined, /* () */0);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMeshRenderer(TestTool$WonderEditor.buildEmptyAppState(/* () */0)));
                              }));
                        return Wonder_jest.test("test logic", (function () {
                                      var triangleFanType = MainEditorMeshRendererTool$WonderEditor.getDrawModeTriangleFanType(/* () */0);
                                      MainEditorMeshRendererTool$WonderEditor.changeMode(triangleFanType, undefined, undefined, undefined, /* () */0);
                                      var meshRenderer = GameObjectTool$WonderEditor.getCurrentGameObjectMeshRenderer(/* () */0);
                                      return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                            return MeshRendererEngineService$WonderEditor.getDrawMode(meshRenderer, param);
                                                          }))), triangleFanType);
                                    }));
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
