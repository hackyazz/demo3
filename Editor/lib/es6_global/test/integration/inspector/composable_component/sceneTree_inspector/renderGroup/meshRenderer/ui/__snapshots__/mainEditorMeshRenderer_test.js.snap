// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainEditor meshRenderer test meshRenderer change drawMode test set drawMode to Lines test snapshot 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      <span
        className=""
      >
        Draw mode
      </span>
    </div>
    <div
      className="item-content"
    >
      <select
        onChange={[Function]}
        value="1"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </div>
  </article>
</article>
`;

exports[`mainEditor meshRenderer test meshRenderer change drawMode test set drawMode to Points test snapshot 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      <span
        className=""
      >
        Draw mode
      </span>
    </div>
    <div
      className="item-content"
    >
      <select
        onChange={[Function]}
        value="0"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </div>
  </article>
</article>
`;

exports[`mainEditor meshRenderer test meshRenderer change drawMode test set drawMode to Triangle_fan test snapshot 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      <span
        className=""
      >
        Draw mode
      </span>
    </div>
    <div
      className="item-content"
    >
      <select
        onChange={[Function]}
        value="6"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </div>
  </article>
</article>
`;
